app/models/tmx_downloading.php

```diff
@@ -47,16 +49,13 @@ function ($e) {
 $empty_TMX = $repos_count == 0;
 
 // Generate the TMX file
+$content = $check['tmx_format'] == 'normal'
+    ? TMX::create($strings, $locale, 'en-US')
+    : TMX::createOmegat($strings, $locale, 'en-US');
 
-if ($check['tmx_format'] == 'normal') {
-    $content = TMX::create($strings, $locale, 'en-US');
-    $target_file = 'mozilla_en-US_' . $locale . '.tmx';
-} else {
-    $content = TMX::createOmegat($strings, $locale, 'en-US');
-    $target_file = 'mozilla_en-US_' . $locale . '_forOmegaT.tmx';
-}
-
-$target_file_path = WEB_ROOT . 'download/' . $target_file;
+$tmx_id = md5($requested_repositories);
+$target_file_name = "mozilla_en-US_{$locale}_{$tmx_id}_{$check['tmx_format']}.tmx";
```

Comment:
Wondering if this is introducing a XSS vector, are we making sure `$check['tmx_format']` can’t contain anything else than "normal" or "omegat"?